src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala

```diff
@@ -51,12 +51,13 @@ class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with Flat
     gdDAO = new MockGoogleDataprocDAO(dataprocConfig, proxyConfig, clusterDefaultsConfig)
     iamDAO = new MockGoogleIamDAO
     samDAO = new MockSamDAO
-    authProvider = new WhitelistAuthProvider(configFactory.getConfig("auth.providerConfig"))
 
     // TODO look into parameterized tests so both provider impls can both be tested
     //serviceAccountProvider = new MockPetServiceAccountProvider(configFactory.getConfig("serviceAccounts.config"))
     serviceAccountProvider = new MockPetsPerProjectServiceAccountProvider(configFactory.getConfig("serviceAccounts.config"))
 
+    authProvider = new SamAuthProvider(configFactory.getConfig("org.broadinstitute.dsde.workbench.leonardo.auth.SamAuthProvider"), serviceAccountProvider)
```

Comment:
Does this work in tests? Do we need a `MockSamAuthProvider`?

Reply:
I don't think we need a `MockSamAuthProvider` (what would it do that alwaysYes/NoProvider or the WhitelistProvider don't already?) but we might want to think about a mock Sam. But we'd have to keep it in sync w/ the real Sam API behaviour, so maybe this just wants to be swat-test only.